pkgs/development/tools/language-servers/dcd/default.nix

```diff
@@ -0,0 +1,74 @@
+{ lib, stdenv, ldc, fetchFromGitHub }:
+
+let
+  dcdVersion = "0.15.2";
+  fetchSource = attrs: fetchFromGitHub (attrs // { name = attrs.repo; });
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dcd";
+  version = dcdVersion;
+
+  srcs = [
+    (fetchSource {
+      owner = "dlang-community";
+      repo = "dcd";
+      rev = "v${dcdVersion}";
+      hash = "sha256-dJ4Ql3P9kPQhQ3ZrNcTAEB5JHSslYn2BN8uqq6vGetY=";
+    })
+    (fetchSource {
+      owner = "dlang-community";
+      repo = "containers";
+      rev = "v0.9.0";
+      hash = "sha256-jJzqUz+/+Yjns4c669WlxLMTWXzG4dkj+ed28+NCEDA=";
+    })
+    (fetchSource {
+      owner = "dlang-community";
+      repo = "libdparse";
+      rev = "v0.23.2";
+      hash = "sha256-r3hrcMELdEKrogsOQTaOtRozsAfRI/2xf4zz8xC9oQA=";
+    })
+    (fetchSource {
+      owner = "msgpack";
+      repo = "msgpack-d";
+      rev = "v1.0.4";
+      hash = "sha256-MU0JtyzxD4vI3cp0HIHhtJ3TtHh3saGSbNdRMRQMa68=";
+    })
+  ];
+
+  patches = [ ./makefile.diff ];
+
+  sourceRoot = ".";
```

Comment:
It'd be more idiomatic to run the build in dcd than having make cd into it.

Reply:
Done. Also changed the makefile diff to use the dependencies in their new locations.